replication/logic/diff/diff.go

```diff
@@ -88,8 +88,20 @@ func SortVersionListByCreateTXGThenBookmarkLTSnapshot(fsvslice []*FilesystemVers
 	return sorted
 }
 
+func FilterBookmarksFromVersionList(fsvslice []*FilesystemVersion) []*FilesystemVersion {
+	fslice := make([]*FilesystemVersion, 0)
```

Comment:
nit: you can use `make(..., 0, len(fsvslice))` to suggest a capacity while still keeping length at 0. Makes for fewer allocations.